packages/babel-types/src/definitions/core.js

```diff
@@ -154,7 +154,7 @@ defineType("CatchClause", {
       validate: assertNodeType("BlockStatement"),
     },
   },
-  aliases: ["Scopable"],
+  aliases: ["Scopable", "BlockParent"],
```

Comment:
I'm not sure about this. Please take a second look at this change.

In `babel-types/src/validators` part of this PR(as shown below), we say flag that the block statement of a CatchClause is not Scopable. And BlockScoped bindings are registered as `scope.getBlockParent().registerBinding`.

So, when registering the `let e` in `catch(e) { let e }`, we have `scope.getBlockParent` where `scope=Catch` and the BlockScope ends up being registered in the wrong scope. So, we include a `BlockParent` alias to the CatchClause.

If you think this should be done in some other way, feel free to suggest. 

The aim is to throw for this code - 

```js
try {} catch (e) { let e }
```